src/v/kafka/server/handlers/fetch.cc

```diff
@@ -522,10 +593,7 @@ static void fill_fetch_responses(
          * Cache fetch metadata
```

Comment:
Are the units being handled correctly in the error case above on lines 586-590? Shouldn't we free an existing units?